app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/StartController.scala

```diff
@@ -45,34 +48,54 @@ class StartController @Inject()(
     with SessionUpdates {
 
   def start(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
-    (request.sessionData.flatMap(_.businessPartnerRecord), request.sessionData.flatMap(_.subscriptionDetails)) match {
-      case (_, Some(_)) =>
-        SeeOther(routes.SubscriptionController.checkYourDetails().url)
+    request.sessionData.flatMap(_.subscriptionStatus) match {
+      case Some(_: SubscriptionReady)         => SeeOther(routes.SubscriptionController.checkYourDetails().url)
+      case Some(_: SubscriptionComplete)      => SeeOther(routes.SubscriptionController.subscribed().url)
+      case Some(SubscriptionMissingData(bpr)) => buildSubscriptionData(Some(bpr))
+      case None                               => buildSubscriptionData(None)
+    }
+  }
 
-      case (maybeBpr, None) =>
-        val result = for {
-          bpr <- maybeBpr.fold(
-                  bprService.getBusinessPartnerRecord(
-                    request.authenticatedRequest.nino,
-                    request.authenticatedRequest.name,
-                    request.authenticatedRequest.dateOfBirth
-                  )
-                )(EitherT.pure(_))
-          subscriptionDetails <- EitherT
-                                  .fromEither[Future](SubscriptionDetails(bpr, request.authenticatedRequest.name))
-                                  .leftMap(Error(_))
-          _ <- EitherT(updateSession(sessionStore, request)(_.copy(subscriptionDetails = Some(subscriptionDetails))))
-        } yield subscriptionDetails
+  def buildSubscriptionData(maybeBpr: Option[BusinessPartnerRecord])(
+    implicit
+    hc: HeaderCarrier,
+    request: RequestWithSessionData[_]
+  ): Future[Result] = {
+    val result = for {
+      bpr <- maybeBpr.fold(
+              bprService.getBusinessPartnerRecord(
+                request.authenticatedRequest.nino,
+                request.authenticatedRequest.name,
+                request.authenticatedRequest.dateOfBirth
+              ))(EitherT.pure(_))
+      maybeSubscriptionDetails <- EitherT.pure(SubscriptionDetails(bpr, request.authenticatedRequest.name))
+      _ <- EitherT(
+            maybeSubscriptionDetails.fold[Future[Either[Error, Unit]]](
+              _ =>
+                updateSession(sessionStore, request)(_.copy(subscriptionStatus = Some(SubscriptionMissingData(bpr)))),
+              subscriptionDetails =>
+                updateSession(sessionStore, request)(
+                  _.copy(subscriptionStatus = Some(SubscriptionReady(subscriptionDetails))))
+            )
+          )
+    } yield maybeSubscriptionDetails
 
-        result.value.map {
-          case Left(e) =>
-            logger.warn("Error while getting subscription details", e)
-            errorHandler.errorResult()
+    result.fold(
+      { e =>
+        logger.warn("Error while getting subscription details", e)
+        errorHandler.errorResult()
+      }, {
+        case Left(missingData) =>
```

Comment:
eventually, it'd be nice to handle all the missing data cases here in a single place - this can probably happen in the future though so this PR isn't too big